xdsl/dialects/riscv.py

```diff
@@ -145,13 +159,21 @@ def register_name(self) -> str:
         return self.data.name
 
     @classmethod
-    def parse_parameter(cls, parser: AttrParser) -> Register:
+    def new_register(cls, name: str | None = None) -> RegisterT:
```

Comment:
This is just the init, right?

Reply:
Actually not. `new_register_type` is the init indeed.

Reply:
Not sure why, but I think my comments got shifted in terms of line numbers.